grpc/src/test/java/com/linecorp/armeria/internal/server/grpc/GrpcDocServiceWithCustomRouteTest.java

```diff
@@ -0,0 +1,134 @@
+                                                    .enableUnframedRequests(true)
+                                                    .build());
+            sb.serviceUnder("/docs",
+                            DocService.builder()
+                                      .exampleRequests(TestServiceGrpc.SERVICE_NAME, "UnaryCall",
+                                                       SimpleRequest.newBuilder()
+                                                                    .setResponseSize(1000)
+                                                                    .setFillUsername(true).build())
+                                      .build()
+            );
+        }
+    };
+
+    @Test
+    void exposeOnlyRoutingMethod() throws Exception {
+        final WebClient client = WebClient.of(server.httpUri());
+        final AggregatedHttpResponse response = client.get("/docs/specification.json").aggregate().join();
+        final JsonNode node = mapper.readValue(response.content().array(), JsonNode.class);
+        final List<JsonNode> methods = ImmutableList.copyOf(node.at("/services/0/methods").elements());
+        assertThat(methods).hasSize(2);
+        assertThat(methods).anyMatch(method -> "EmptyCall".equals(method.get("name").textValue()));
+        assertThat(methods).anyMatch(method -> "UnaryCall".equals(method.get("name").textValue()));
+    }
+
+    @Test
+    void mixUnframedWithFramed() throws Exception {
+        final WebClient client = WebClient.of(unifiedServer.httpUri());
+        final AggregatedHttpResponse response = client.get("/docs/specification.json").aggregate().join();
+
+        final JsonNode node = mapper.readValue(response.content().array(), JsonNode.class);
+        final List<JsonNode> methods = ImmutableList.copyOf(node.at("/services/0/methods").elements());
+        assertThat(methods).hasSize(testService.bindService().getMethods().size());
+
+        final MediaType protobuf = MediaType.PROTOBUF.withParameter("protocol", "gRPC");
+        final MediaType json = MediaType.JSON_UTF_8.withParameter("protocol", "gRPC");
+        for (JsonNode method : methods) {
+            final JsonNode endpoints = method.get("endpoints");
+            final String methodName = method.get("name").textValue();
+            if ("EmptyCall".equals(methodName) || "UnaryCall".equals(methodName)) {
+                assertThat(validateEndpointMimeType(endpoints, protobuf)).isTrue();
+                assertThat(validateEndpointMimeType(endpoints, json)).isTrue();
+            } else {
+                assertThat(validateEndpointMimeType(endpoints, protobuf)).isFalse();
+                assertThat(validateEndpointMimeType(endpoints, json)).isFalse();
+            }
+        }
+    }
+
+    private static boolean validateEndpointMimeType(JsonNode endpoint, MediaType mediaType) {
+        return ImmutableList
+                       .copyOf(endpoint.elements()).stream()
```

Comment:
Global comment: `ImmutableList.copyOf().stream()` could be just `Streams.stream(iterator)` (Guava)